SimTKcommon/tests/TestPlugin.cpp

```diff
@@ -190,6 +190,488 @@ void testPathname() {
+    SimTK_TEST(directory == dir && fileName == "myFileName" && extension == ".ext" && dontApplySearchPath);
+    pathname = "junk";
+    pathname = Pathname::getAbsolutePathnameUsingSpecifiedWorkingDirectory(swd, path);
+    SimTK_TEST(pathname == dir + "myFileName.ext");
+
+    swd = "/specified"; path = "topdir/seconddir/myFileName.ext";
+    dir = curDrive + sep + "specified" + sep + "topdir" + sep + "seconddir" + sep;
+    directory = fileName = extension = "junk"; dontApplySearchPath = false;
+    Pathname::deconstructPathnameUsingSpecifiedWorkingDirectory(swd, path, directory, fileName, extension, dontApplySearchPath);
+    SimTK_TEST(directory == dir && fileName == "myFileName" && extension == ".ext" && dontApplySearchPath);
+    pathname = "junk";
+    pathname = Pathname::getAbsolutePathnameUsingSpecifiedWorkingDirectory(swd, path);
+    SimTK_TEST(pathname == dir + "myFileName.ext");
+
+    ///
+
+    swd = "Y:specified"; path = "X:/topdir/seconddir/myFileName.ext";
+    dir = "x:" + sep + "topdir" + sep + "seconddir" + sep;
+    directory = fileName = extension = "junk"; dontApplySearchPath = false;
+    Pathname::deconstructPathnameUsingSpecifiedWorkingDirectory(swd, path, directory, fileName, extension, dontApplySearchPath);
+    SimTK_TEST(directory == dir && fileName == "myFileName" && extension == ".ext" && dontApplySearchPath);
+    pathname = "junk";
+    pathname = Pathname::getAbsolutePathnameUsingSpecifiedWorkingDirectory(swd, path);
+    SimTK_TEST(pathname == dir + "myFileName.ext");
+
+    swd = "Y:specified"; path = "/topdir/seconddir/myFileName.ext";
+    dir = "y:" + sep + "topdir" + sep + "seconddir" + sep;
+    directory = fileName = extension = "junk"; dontApplySearchPath = false;
+    Pathname::deconstructPathnameUsingSpecifiedWorkingDirectory(swd, path, directory, fileName, extension, dontApplySearchPath);
+    SimTK_TEST(directory == dir && fileName == "myFileName" && extension == ".ext" && dontApplySearchPath);
+    pathname = "junk";
+    pathname = Pathname::getAbsolutePathnameUsingSpecifiedWorkingDirectory(swd, path);
+    SimTK_TEST(pathname == dir + "myFileName.ext");
+
+    swd = "Y:specified"; path = "X:topdir/seconddir/myFileName.ext";
+    dir = "y:" + sep + "specified" + sep + "topdir" + sep + "seconddir" + sep;
+    directory = fileName = extension = "junk"; dontApplySearchPath = false;
+    Pathname::deconstructPathnameUsingSpecifiedWorkingDirectory(swd, path, directory, fileName, extension, dontApplySearchPath);
+    SimTK_TEST(directory == dir && fileName == "myFileName" && extension == ".ext" && dontApplySearchPath);
+    pathname = "junk";
+    pathname = Pathname::getAbsolutePathnameUsingSpecifiedWorkingDirectory(swd, path);
+    SimTK_TEST(pathname == dir + "myFileName.ext");
+
+    swd = "Y:specified"; path = "./topdir/seconddir/myFileName.ext";
+    dir = "y:" + sep + "specified" + sep + "topdir" + sep + "seconddir" + sep;
```

Comment:
Might be better to use "C:" and "D:" drives since most Windows machines have both.